lib/errors.js

```diff
@@ -285,6 +285,9 @@ internals.annotate = function (stripColorCodes) {
                         delete ref[replacement];
                     }
                     else {
+                        if (typeof ref === 'string') {
```

Comment:
@Marsup I tried this:

```js
            const schema = {
                a: Joi.object({
                    b: Joi.object({
                        c: Joi.string()
                    })
                })
            };

            const input = {
                a: '{"b":{"d":"string"}}'
            };

            expect(() => {

                Joi.attempt(input, schema);
            }).to.throw(/\"d\" is not allowed/);
            done();
```

and that:

```js
            const schema = {
                a: Joi.object({
                    b: Joi.object({
                        c: Joi.object({
                            d: Joi.string()
                        })
                    })
                })
            };

            const input = {
                a: '{"b":{"c":{"d":{}}}}'
            };

            expect(() => {

                Joi.attempt(input, schema);
            }).to.throw(/\"d\" must be a string/);
            done();
```

Both cases are not failing without any changes to the fix.